src/Pkb/Parsobober.Pkb.Relations/Implementations/ModifiesRelation.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.Extensions.Logging;
+using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Accessors;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Pkb.Relations.Implementations;
+
+public class ModifiesRelation(
+    ILogger<ParentRelation> logger,
+    IProgramContextAccessor programContext
+) : IModifiesCreator
+{
+    /// <summary>
+    /// Stores modifies relation between statement(assign) and variable.
+    /// </summary>
+    /// <remarks>[statement(assign) line number,  variable name].</remarks>
+    private readonly Dictionary<int, string> _statementModifiesDictionary = new();
+
+    /// <summary>
+    /// Stores modifies relation between container statement(while, if) and variable list
+    /// </summary>
+    /// <remarks>[statement(while, if) line number,  list of variable names].</remarks>
+    private readonly Dictionary<int, List<string>> _containerModifiesDictionary = new();
+
+    public void SetModifies(TreeNode modifier, TreeNode variable)
+    {
+        // Check types
+        if (modifier.Type.IsStatement() == false || variable.Type != EntityType.Variable)
+        {
+            logger.LogError(
+                "Statement modifies relation can only be established between statement and variable node. ({modifier} => {variable})",
+                modifier, variable);
+
+            throw new ArgumentException(
+                $"At least one of provided nodes type: {modifier.Type}, {variable.Type} is different than any of required: {EntityType.Statement} types or {EntityType.Variable}.");
```

Comment:
Is there any reason why we can't tell which one has invalid type and pass it as a message?